docs/config.md

```diff
@@ -660,28 +660,77 @@ export PROMREGATOR_AUTHENTICATOR_BASIC_PASSWORD=myPassword
+
+#### Option "promregator.authenticator.oauth2xsuaaBasic.client_id" (mandatory, if using promregator.authenticator.type=OAuth2XSUAABasic)
 Specifies the client identifier (a.k.a. "client_id") which shall be used during the OAuth2 request based on the Grant Type Client Credentials flow in case of global authentication.
 
-#### Option "promregator.authenticator.oauth2xsuaa.client_secret" (mandatory, if using promregator.authenticator.type=OAuth2XSUAA)
+#### Option "promregator.authenticator.oauth2xsuaaCertificate.client_id" (mandatory, if using promregator.authenticator.type=OAuth2XSUAACertificate)
+Specifies the client identifier (a.k.a. "client_id") which shall be used during the OAuth2 request based on the Grant Type Client Credentials flow in case of global authentication.
+
+#### Option "promregator.authenticator.oauth2xsuaaCertificate.client_certificates" (mandatory if using promregator.authenticator.type=OAuth2XSUAACertificate)
+Specifies the certificate chain which shall be used during the OAuth2 request.
+
+#### Option "promregator.authenticator.oauth2xsuaa.client_secret" (mandatory, if using promregator.authenticator.type=OAuth2XSUAA, *deprecated*)
+Specifies the client secret (a.k.a. "client_secret") which shall be used during the OAuth2 request based on the Grant Type Client Credentials flow in case of global authentication. Deprecated, please use `promregator.authenticator.oauth2xsuaaBasic.client_secret` instead.
+
+*WARNING!*
+Due to security reasons, it is *neither* recommended to store this value in your YAML file, nor to put it into the command line when starting Promregator.
+Instead it is suggested to set the corresponding environment variables `PROMREGATOR_AUTHENTICATOR_OAUTH2XSUAA_CLIENT_SECRET` when starting the application.
+
+Example:
+
+```bash
+export PROMREGATOR_AUTHENTICATOR_OAUTH2XSUAA_CLIENT_SECRET=myClientSecret
+java -Dspring.config.location=file:/path/to/your/myconfig.yaml -jar promregator-0.0.1-SNAPSHOT.jar
+```
+
+#### Option "promregator.authenticator.oauth2xsuaaBasic.client_secret" (mandatory, if using promregator.authenticator.type=OAuth2XSUAABasic)
 Specifies the client secret (a.k.a. "client_secret") which shall be used during the OAuth2 request based on the Grant Type Client Credentials flow in case of global authentication.
 
+*WARNING!*
+Due to security reasons, it is *neither* recommended to store this value in your YAML file, nor to put it into the command line when starting Promregator.
+Instead it is suggested to set the corresponding environment variables `PROMREGATOR_AUTHENTICATOR_OAUTH2XSUAA_BASIC_CLIENT_SECRET` when starting the application.
+
+Example:
+
+```bash
+export PROMREGATOR_AUTHENTICATOR_OAUTH2XSUAA_BASIC_CLIENT_SECRET=myClientSecret
```

Comment:
Is `PROMREGATOR_AUTHENTICATOR_OAUTH2XSUAA_BASIC_CLIENT_SECRET` really correct? I would have expected `PROMREGATOR_AUTHENTICATOR_OAUTH2XSUAABASIC_CLIENT_SECRET`.

Reply:
It works with these environment variables which are also contained in the docu like this:

- `PROMREGATOR_AUTHENTICATOR_OAUTH2XSUAA_BASIC_CLIENT_SECRET`
- `PROMREGATOR_AUTHENTICATOR_OAUTH2XSUAA_CLIENT_SECRET`
- `PROMREGATOR_AUTHENTICATOR_OAUTH2XSUAA_CERTIFICATE_CLIENT_KEY`

It works with the underscore between `OAUTH2XSUAA` and `BASIC`.